ocaml/xapi/message_forwarding.ml

```diff
@@ -3546,6 +3546,25 @@ functor
             Client.Host.get_server_certificate rpc session_id host
         )
 
+      let refresh_server_certificate ~__context ~host =
```

Comment:
We probably want a lock to prevent concurrent refreshes

Reply:
Maybe we want to error if a refresh is already in progress

Reply:
your call to `with_pool_operation` should error if a refresh is already in progress